plugins/eventlog/eventlog.c

```diff
@@ -0,0 +1,379 @@
+        ns_rr  rr;
+        qdcount = ns_msg_count(msg, ns_s_qd);
+
+        if (qdcount > 0 && 0 == (err = ns_parserr(&msg, ns_s_qd, 0, &rr)) && ns_rr_class(rr) == 1) {
+            fprintf(out, "%s %s\n",
+                p_type(ns_rr_type(rr)),
+                ns_rr_name(rr));
+        }
+        if (err < 0) {
+            if (tcpstate_getcurr && tcpstate_reset)
+                tcpstate_reset(tcpstate_getcurr(), "");
+        }
+        return;
+    }
+
+    /*
+     * IP Stuff
+     */
+    fprintf(out, "src=%s spt=%u ", ia_str(from), sport);
+    fprintf(out, "dst=%s dpt=%u ", ia_str(to), dport);
+    switch(proto) {
+       case 17:
+          fprintf(out, "proto=UDP");
+          break;
+       case 6:
+          fprintf(out, "proto=TCP");
+          break;
+       default:
+          fprintf(out, "proto=%hhu", proto);
+          break;
+    }
+    
+    int    rrnum, ancount, qdcount, err = 0;
+    ns_rr  rr;
+    char  *delim;
+
+    /*
+     * DNS Header
+     */
+    fprintf(out, " mid=%u", ns_msg_id(msg));
+    fprintf(out, " op=%u", ns_msg_getflag(msg, ns_f_opcode));
+    fprintf(out, " fl=|");
+    if (ns_msg_getflag(msg, ns_f_qr))
+        fprintf(out, "QR|");
+    if (ns_msg_getflag(msg, ns_f_aa))
+        fprintf(out, "AA|");
+    if (ns_msg_getflag(msg, ns_f_tc))
+        fprintf(out, "TC|");
+    if (ns_msg_getflag(msg, ns_f_rd))
+        fprintf(out, "RD|");
+    if (ns_msg_getflag(msg, ns_f_ra))
+        fprintf(out, "RA|");
+    if (ns_msg_getflag(msg, ns_f_ad))
+        fprintf(out, "AD|");
+    if (ns_msg_getflag(msg, ns_f_cd))
+        fprintf(out, "CD|");
+    switch(ns_msg_getflag(msg, ns_f_rcode)) {
+        case ns_r_noerror:
+          fprintf(out, " rc=OK");
+          break;
+        case ns_r_nxdomain:
+          fprintf(out, " rc=NXDOMAIN");
+          break;
+        case ns_r_servfail:
+          fprintf(out, " rc=SRVFAIL");
+          break;
+        default:
+          fprintf(out, " rc=%u", ns_msg_getflag(msg, ns_f_rcode));
+          break;
+    }
+
+    qdcount = ns_msg_count(msg, ns_s_qd);
+    if (qdcount > 0 && 0 == (err = ns_parserr(&msg, ns_s_qd, 0, &rr))) {
+        fprintf(out, " cl=%s tp=%s name=%s",
+            p_class(ns_rr_class(rr)),
+            p_type(ns_rr_type(rr)),
+            ns_rr_name(rr));
+    }
+    if (err < 0) {
+        fprintf(out, " **ERROR parsing response record**\n");
+        if (tcpstate_getcurr && tcpstate_reset)
+            tcpstate_reset(tcpstate_getcurr(), "");
+        return;
+    }
+
+    /* output the query answers */
+    delim = " ans=";
+    for(rrnum = 0; rrnum < ns_msg_count(msg, ns_s_an); rrnum++) {
+        if (0 == (err = ns_parserr(&msg, ns_s_an, rrnum, &rr))) {
+          /* If the answer is an IP address, output it. */
+            if((0 == strncmp(p_type(ns_rr_type(rr)), "A", 2)) ||
+               (0 == strncmp(p_type(ns_rr_type(rr)), "AAAA", 5))) {
+                fprintf(out, "%s", delim);
+                delim=", ";
```

Comment:
Wouldn't it be better to not have a space in the delimiter here?

So that the `ans=` list would be just comma separated and easily split from rest of the output by space.

Reply:
Certainly. Done.